src/middlewared/middlewared/alert/source/audit.py

```diff
@@ -0,0 +1,52 @@
+from datetime import timedelta
+import logging
+from middlewared.alert.base import AlertClass, AlertCategory, Alert, AlertLevel, AlertSource, SimpleOneShotAlertClass
+from middlewared.alert.schedule import IntervalSchedule
+
+log = logging.getLogger("audit_check_alertmod")
+
+
+# -------------- OneShot Alerts ------------------
+class AuditBackendSetupAlertClass(AlertClass, SimpleOneShotAlertClass):
+    category = AlertCategory.AUDIT
+    level = AlertLevel.ERROR
+    title = "Audit Service Backend Failed"
+    text = "Audit service failed backend setup: %(service)s. See /var/log/middlewared.log"
+
+
+class AuditSetupAlertClass(AlertClass, SimpleOneShotAlertClass):
+    category = AlertCategory.AUDIT
+    level = AlertLevel.ERROR
+    title = "Audit Service Setup Failed"
+    text = "Audit service failed to complete setup. See /var/log/middlewared.log"
+
+
+# --------------- Monitored Alerts ----------------
+class AuditServiceHealthAlertClass(AlertClass):
+    category = AlertCategory.AUDIT
+    level = AlertLevel.ERROR
+    title = "Audit Service Health Failure"
+    text = "Failed to perform audit query: %(verrs)s"
+
+
+class AuditServiceHealthAlertSource(AlertSource):
+    '''
+    Run simple query every 20 minutes as a heath check
+    '''
+    schedule = IntervalSchedule(timedelta(minutes=20))
+    run_on_backup_node = False
+
+    async def check(self):
+        try:
+            await self.middleware.call(
+                'audit.query', {
+                    "query-filters": [["event", "=", "AUTHENTICATION"]],
+                    "query-options": {"offset": -1}
```

Comment:
Did you test how expensive this is when we have a couple of hundred thousand entries?

Reply:
I think it would be better to short-circuit query with `{"get": true}` rather than specifying offsets (pending tests).

Reply:
What does negative offset even mean?

Reply:
Negative offset means: Get the last entry.